client/asset/dcr/rpcwallet.go

```diff
@@ -1030,10 +1055,23 @@ func (w *rpcWallet) PurchaseTickets(ctx context.Context, n int, _, _ string) ([]
 	return tickets, nil
 }
 
+var oldSPVWalletErr = errors.New("wallet is an older spv wallet")
+
 // Tickets returns active tickets.
 func (w *rpcWallet) Tickets(ctx context.Context) ([]*asset.Ticket, error) {
-	const includeImmature = true
-	// GetTickets only works for clients with a dcrd backend.
+	return w.tickets(ctx, true)
+}
+
+func (w *rpcWallet) tickets(ctx context.Context, includeImmature bool) ([]*asset.Ticket, error) {
+	w.ticketCache.Lock()
+	defer w.ticketCache.Unlock()
+	if w.ticketCache.stamp.Add(ticketCacheExpiry).After(time.Now()) {
+		return w.ticketCache.tickets, nil
+	}
```

Comment:
Sometimes this was returning tickets then none if called close together. I guess a cache is alright to have?